backend/btrixcloud/users.py

```diff
@@ -507,16 +512,37 @@ async def _update_password(self, user: User, new_password: str) -> None:
         """Update hashed_password for user, overwriting previous password hash
 
         Internal method, use change_password() for password verification first
+
+        Method also ensures user is not locked after password change
         """
         await self.validate_password(new_password)
         hashed_password = get_password_hash(new_password)
         if hashed_password == user.hashed_password:
             return
         user.hashed_password = hashed_password
         await self.users.find_one_and_update(
-            {"id": user.id}, {"$set": {"hashed_password": hashed_password}}
+            {"id": user.id},
+            {"$set": {"hashed_password": hashed_password, "locked": False}},
         )
 
+    async def set_failed_logins(self, user: User, count: int) -> None:
+        """Set consecutive failed login attempts for user"""
+        await self.users.find_one_and_update(
+            {"id": user.id}, {"$set": {"failed_logins": count}}
+        )
+
+    async def inc_failed_logins(self, user: User) -> None:
+        """Inc consecutive failed login attempts for user by 1"""
+        count = await self.get_failed_logins_count(user)
+        await self.set_failed_logins(user, count + 1)
```

Comment:
Should use mongo `$inc` operator instead.

Maybe the two functions are `inc_failed_logins` which uses inc and `clear_failed_logins` which resets it to 0?